spec/models/enterprise_relationship_spec.rb

```diff
@@ -255,5 +255,101 @@
         end
       end
     end
+    describe "updating order cycles" do
+      let(:hub) { create(:distributor_enterprise) }
+      let(:producer) { create(:supplier_enterprise) }
+      let(:order_cycle) { create(:simple_order_cycle) }
+      let(:some_other_producer) { create(:supplier_enterprise) }
+
+      context "when order_cycle permission is present" do
+        let!(:er) {
+          create(:enterprise_relationship, child: hub, parent: producer,
+                                           permissions_list: [
+                                             :add_to_order_cycle,
+                                             :create_variant_overrides
+                                           ] )
+        }
+        let!(:incoming_external_exchange) {
+          order_cycle.exchanges.create! sender: producer, receiver: hub, incoming: true
+        }
+        let!(:other_external_exchange) {
+          order_cycle.exchanges.create! sender: some_other_producer, receiver: hub, incoming: true
+        }
+        let!(:incoming_internal_exchange) {
+          order_cycle.exchanges.create! sender: hub, receiver: hub, incoming: true
+        }
+        let!(:outgoing_internal_exchange) {
+          order_cycle.exchanges.create! sender: hub, receiver: hub, incoming: false
+        }
+        let!(:variant) { create(:variant) }
+        let!(:some_other_variant) { create(:variant) }
+        let!(:incoming_external_variant) {
+          incoming_external_exchange.exchange_variants.create!(
+            exchange: incoming_external_exchange, variant: variant
+          )
+        }
+        let!(:incoming_internal_only_variant) {
+          incoming_internal_exchange.exchange_variants.create!(
+            exchange: incoming_internal_exchange, variant: some_other_variant
+          )
+        }
+        let!(:outgoing_internal_variant) {
+          outgoing_internal_exchange.exchange_variants.create!(
+            exchange: outgoing_internal_exchange, variant: variant
+          )
+        }
+        let!(:outgoing_internal_only_variant) {
+          outgoing_internal_exchange.exchange_variants.create!(
+            exchange: outgoing_internal_exchange, variant: some_other_variant
+          )
+        }
+
+        # We need to destroy the exchange variants on all order cycles related to the ER if
+        # 'add_to_order_cycle' permission is removed. If they are left on the order cycle, the
+        # Taxons of the variants will still appear on the /shops page, despite the hub not
+        # actually offering the variants anymore.
+        context "removing exchanges and exchange variants" do
+          context "when the enterprise relationship is destroyed" do
+            before { er.destroy }
```

Comment:
I personally would do the destroy within the test. And I would probably include it within an expectation:
```rb
expect {
  er.destroy
}.to change {
  ExchangeVariant.count
}.by(-2)
```